src/BackupDestination/Backup.php

```diff
@@ -54,6 +77,10 @@ public function stream()
 
     public function delete()
     {
+        $this->exists = null;
```

Comment:
Do we really need to set these to `null`. I assume that when calling `delete`, this instance won't be used anymore by the rest of the code.

Reply:
It's a precaution, I guess.

Since the `size()` call contains this snippet
```
        if (! $this->exists()) {
            return 0;
```

I'm lead to assume there have been occurences where `Backup` instances point to deleted files.